clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerCoordinator.java

```diff
@@ -692,10 +693,18 @@ private void validateCooperativeAssignment(final Map<String, List<TopicPartition
     }
 
     @Override
-    protected void onJoinPrepare(int generation, String memberId) {
+    protected boolean onJoinPrepare(int generation, String memberId) {
         log.debug("Executing onJoinPrepare with generation {} and memberId {}", generation, memberId);
-        // commit offsets prior to rebalance if auto-commit enabled
-        maybeAutoCommitOffsetsSync(time.timer(rebalanceConfig.rebalanceTimeoutMs));
+        boolean onJoinPrepareAsyncCommitSucceeded;
+        try {
+            // async commit offsets prior to rebalance if auto-commit enabled
+            onJoinPrepareAsyncCommitSucceeded = maybeAutoCommitOffsetsAsync();
+        } catch (Exception e) {
+            onJoinPrepareAsyncCommitFuture = null;
+            onJoinPrepareAsyncCommitSucceeded = true;
```

Comment:
If the exception is of non-retryable type, decide to abandon to asyncCommit offsets

Reply:
So, 2 comments here:
1. If we return `true`, we won't do `onJoinPrepare` on next poll, and the offset commit will be skipped. Is that expected? 
2. the returned `future` null means no commit operated (since no auto commit). Could you add a comment here to explain the case?
3. The variable name `onJoinPrepareAsyncCommitSucceeded` might be better to change to `onJoinPrepareAsyncCommitCompleted` ? WDYT?

Reply:
1. Yes.
If (!autoCommitEnabled || coordinatorUnknown()), future will be null, I think in this case, we can give up committing offsets for rebalance
2&3. sure